stdlib/public/Differentiation/Differentiable.swift

```diff
@@ -59,12 +82,18 @@ public extension Differentiable where TangentVector == Self {
 
 // SWIFT_ENABLE_TENSORFLOW
 public extension Differentiable {
-  // This is a temporary solution that allows us to add `zeroTangentVector`
-  // without implementing derived conformances. This property is marked
-  // unavailable because it will produce incorrect results when tangent vectors
-  // depend on instance properties of `self`.
-  // FIXME: Implement derived conformance and remove this default
+  // This is a temporary solution enabling the addition of
+  // `zeroTangentVectorInitializer` without implementing derived conformances.
+  // This property will produce incorrect results when tangent vectors depend
+  // on instance-specific information from `self`.
+  // FIXME: Implement derived conformances and remove this default
   // implementation.
-  var zeroTangentVector: TangentVector { .zero }
+  var zeroTangentVectorInitializer: () -> TangentVector {
+    { TangentVector.zero }
+  }
+
+  /// A tangent vector initialized using `zeroTangentVectorInitializer`.
+  /// `move(along: zeroTangentVector)` should not modify `self`.
+  var zeroTangentVector: TangentVector { zeroTangentVectorInitializer() }
```

Comment:
Unfortunately, there's a availability warning when compiling the standard library:

```
Differentiable.swift:97:42: warning: 'zeroTangentVectorInitializer' is deprecated: `zeroTangentVectorInitializer` derivation has not been implemented; do not use this property
  var zeroTangentVector: TangentVector { zeroTangentVectorInitializer() }
                                         ^
```

I think this is acceptable for now.